### PR TITLE
Emit focus events on window selection changes

### LINKED ISSUE #140: Selecting different window in Emacs does not trigger focus inside Ghostel
As an example, when Claude Code loses focus, it hides its cursor. If I unfocus Emacs entirely, Ghostel correctly signals to Claude Code (or whatever process is running) that there is no focus. However, the same does not happen when selecting a different window inside of Emacs. My expecation would be that selecting another window would trigger the same behavior in Ghostel as selecting an entirely different application.

### PR DESCRIPTION
**Pull request by dakra**
Closes #140.

## Summary

- Emit VT focus events (CSI `\e[I` / `\e[O`) when the user selects a different window inside Emacs, not just when the whole frame blurs. Child programs like Claude Code, vim, or btop that enable mode 1004 now correctly see focus-out/focus-in on intra-frame window switches and minibuffer activation.
- Logical focus is computed per ghostel buffer as "frame is focused AND buffer is in that frame's selected window." Hooks into `window-selection-change-functions` and `window-buffer-change-functions` in addition to `after-focus-change-function`; a buffer-local `ghostel--focus-state` dedups so only real transitions emit events.
- Fix a latent sentinel bug: `(remove-function after-focus-change-function #'ghostel--focus-change)` was called on every process exit, globally removing the hook and breaking focus reporting for all other live ghostel buffers. The handler's existing guards already no-op after exit, so the removal was both wrong and unnecessary.

## Test plan

- [x] `make -j4 all` passes (131 elisp + 78 native + 32 evil tests, lint clean)
- [x] Five new pure-elisp ERT tests added:
  - `ghostel-test-focus-window-selection` — window selection flips focus state
  - `ghostel-test-focus-dedup` — repeated calls with unchanged state are no-ops
  - `ghostel-test-focus-two-ghostel-buffers` — only the selected ghostel buffer is focused
  - `ghostel-test-focus-frame-blur` — frame-level blur/refocus still works
  - `ghostel-test-focus-minibuffer` — minibuffer activation drives focus-out
- [ ] Manual: run `claude` in a ghostel buffer, split window, select the other window — cursor hides. Reselect — cursor reappears.

### CODE REVIEW

File: ghostel.el [hunk 1]
```diff
@@ -2127,17 +2127,40 @@ Call this after changing the Emacs theme so terminals match."
...
       (with-current-buffer buf
         (when (and (derived-mode-p 'ghostel-mode)
                    ghostel--term
                    ghostel--process
                    (process-live-p ghostel--process))
-          (ghostel--focus-event ghostel--term focused))))))
+          (let ((focused (and (ghostel--buffer-focused-p buf) t)))
+            (unless (eq focused ghostel--focus-state)
+              (setq ghostel--focus-state focused)
+              (ghostel--focus-event ghostel--term focused))))))))
```
[the remark this answers is not in the document]
Comment (dakra) [reply ghostel.el]: Good catch — applied in 7e48709. `ghostel--focus-state` now updates only when `ghostel--focus-event` returns non-nil, so enabling mode 1004 after a focus change does not dedup away the first real event. Added `ghostel-test-focus-skips-state-update-when-1004-off` to cover it.

[the remark this answers is not in the document]
Comment (dakra) [reply ghostel.el]: Declining. The hook fires on user-initiated window/buffer changes (not a tight loop), `derived-mode-p` short-circuits immediately on non-ghostel buffers, and `(buffer-list)` is O(small) on any realistic session. Tracking a separate global list would add buffer-kill bookkeeping and edge cases (mode changes, buffer renaming) for negligible real-world savings. Keeping the simpler implementation.

File: test/ghostel-test.el [hunk 2]
```diff
@@ -1228,6 +1228,176 @@ the reply waits for the redraw timer."
...
+          (delete-other-windows)
+          (switch-to-buffer buf)
+          (ghostel--focus-change)
+          (should (equal (car events) (cons buf t)))
+          ;; Simulate minibuffer activation by selecting the minibuffer window.
+          (let ((mb-win (minibuffer-window)))
+            (select-window mb-win)
+            (setq events nil)
+            (ghostel--focus-change)
+            (should (equal (car events) (cons buf nil)))))
```
[the remark this answers is not in the document]
Comment (dakra) [reply test/ghostel-test.el:1781]: Declining. The current test passes reliably in batch (verified across multiple runs). `select-window` on an inactive minibuffer window is allowed and sufficient here — the assertion is purely about logical focus (ghostel window is no longer the frame-selected-window → focus-out), which does not require the minibuffer to be read-active. The suggested `read-from-minibuffer` + setup-hook pattern would also couple the test to minibuffer-exit timing and risk stdin behavior in batch mode.
